assembly/buffer/index.ts

```diff
@@ -1,5 +1,27 @@
+import { BLOCK_MAXSIZE } from "rt/common";
+import { E_INVALIDLENGTH, E_INDEXOUTOFRANGE } from "util/error";
+
 export class Buffer extends Uint8Array {
   constructor(size: i32) {
     super(size);
   }
+
+  public static alloc(size: i32): Buffer {
+    return new Buffer(size);
+  }
+
+  @unsafe public static allocUnsafe(size: i32): Buffer {
+    // Node throws an error if size is less than 0
+    if (u32(size) > BLOCK_MAXSIZE) throw new RangeError(E_INVALIDLENGTH);
+    let buffer = __alloc(size, idof<ArrayBuffer>());
+    // This retains the pointer to the result Buffer.
+    let result = changetype<Buffer>(__alloc(offsetof<Buffer>(), idof<Buffer>()));
+    // @ts-ignore: This retains the pointer to the result underlying ArrayBuffer
```

Comment:
Might be possible to avoid the ts-ignores by means of `import { Uint8Array } from "typedarray"`, that is if the IDE picks this up properly.

Reply:
Definitely possible. All set!